source/Nevermore/Advanced/WriteTransaction.cs

```diff
@@ -216,7 +230,7 @@ public string AllocateId(string tableName, string idPrefix)
             return AllocateId(tableName, key => $"{idPrefix}-{key}");
```

Comment:
This overload is a problem. Ideally you shouldn't be able to Allocate an Id for an Identity column (the db is doing it for you) and it should throw an exception, however you only receive a string name, so we either have to look up the mapping (which we are already doing above) or we just have to let the code fail badly.

I would be in favour of removing the public API `string AllocateId(string tableName, string idPrefix)` and only allowing public API's that take something that has a DocumentMap, where we can change this.

How often are you allocating Id's for a table that you don't have a document map for... seems weird

Reply:
Octopus server does exactly this to allocate Tag Ids. They are technically a child of TagSet, but the Ids are issued as though it was a table of its own. Agreed it is an edge case usage, but it is used.